server/src/main/java/org/activityinfo/server/endpoint/rest/DatabaseResource.java

```diff
@@ -147,4 +172,245 @@ public Response createFormFromXForm(@Context UriInfo uri, XForm xForm) {
                 .build())
                 .build();
     }
+
+    private Database getDatabase() {
+        return entityManagerProvider.get().find(Database.class, databaseId);
+    }
+
+    private Maybe<User> getUserByEmail(String userEmail) throws NoResultException {
```

Comment:
Use java.util.Optional here instead of Maybe. We use the Maybe type for results which could have *one* of the following states: visible, forbidden, non-existant,or deleted. Here the user is either present or not.